# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1493184769 14400
#      Wed Apr 26 01:32:49 2017 -0400
# Node ID d6e81498e7fe7ef452ad268e46bced902841a913
# Parent  3f1bf237908b75b41feab4bbd2c8790cca98efe3
remove some unnecessary forward declarations

* pt-binop.h, pt-cbinop.h, pt-colon.h, pt-loop.h, pt-unop.h:
Delete unnecessary forward declaration of octave_lvalue.

diff --git a/libinterp/parse-tree/pt-binop.h b/libinterp/parse-tree/pt-binop.h
--- a/libinterp/parse-tree/pt-binop.h
+++ b/libinterp/parse-tree/pt-binop.h
@@ -24,17 +24,16 @@ along with Octave; see the file COPYING.
 #define octave_pt_binop_h 1
 
 #include "octave-config.h"
 
 #include <string>
 
 class octave_value;
 class octave_value_list;
-class octave_lvalue;
 
 #include "ov.h"
 #include "pt-exp.h"
 #include "pt-walk.h"
 #include "symtab.h"
 
 namespace octave
 {
diff --git a/libinterp/parse-tree/pt-cbinop.h b/libinterp/parse-tree/pt-cbinop.h
--- a/libinterp/parse-tree/pt-cbinop.h
+++ b/libinterp/parse-tree/pt-cbinop.h
@@ -24,17 +24,16 @@ along with Octave; see the file COPYING.
 #define octave_pt_cbinop_h 1
 
 #include "octave-config.h"
 
 #include <string>
 
 class octave_value;
 class octave_value_list;
-class octave_lvalue;
 
 #include "ov.h"
 #include "pt-binop.h"
 #include "pt-walk.h"
 #include "symtab.h"
 
 namespace octave
 {
diff --git a/libinterp/parse-tree/pt-colon.h b/libinterp/parse-tree/pt-colon.h
--- a/libinterp/parse-tree/pt-colon.h
+++ b/libinterp/parse-tree/pt-colon.h
@@ -24,17 +24,16 @@ along with Octave; see the file COPYING.
 #define octave_pt_colon_h 1
 
 #include "octave-config.h"
 
 #include <string>
 
 class octave_value;
 class octave_value_list;
-class octave_lvalue;
 
 #include "pt-exp.h"
 #include "pt-walk.h"
 #include "symtab.h"
 
 namespace octave
 {
   // Colon expressions.
diff --git a/libinterp/parse-tree/pt-loop.h b/libinterp/parse-tree/pt-loop.h
--- a/libinterp/parse-tree/pt-loop.h
+++ b/libinterp/parse-tree/pt-loop.h
@@ -21,17 +21,16 @@ along with Octave; see the file COPYING.
 */
 
 #if ! defined (octave_pt_loop_h)
 #define octave_pt_loop_h 1
 
 #include "octave-config.h"
 
 class octave_value;
-class octave_lvalue;
 
 #include "comment-list.h"
 #include "pt-cmd.h"
 #include "pt-walk.h"
 #include "symtab.h"
 
 class jit_info;
 
diff --git a/libinterp/parse-tree/pt-unop.h b/libinterp/parse-tree/pt-unop.h
--- a/libinterp/parse-tree/pt-unop.h
+++ b/libinterp/parse-tree/pt-unop.h
@@ -24,17 +24,16 @@ along with Octave; see the file COPYING.
 #define octave_pt_unop_h 1
 
 #include "octave-config.h"
 
 #include <string>
 
 class octave_value;
 class octave_value_list;
-class octave_lvalue;
 
 #include "pt-exp.h"
 #include "pt-walk.h"
 #include "symtab.h"
 
 namespace octave
 {
   // Unary expressions.
